src/providers/fiat.rs

```diff
@@ -21,10 +21,10 @@ pub struct FiatInfoProvider {
 }
 
 impl FiatInfoProvider {
-    pub fn new(context: &RequestContext) -> Self {
+    pub fn new(context: &RequestContext, chain_id: &str) -> Self {
```

Comment:
Hmmm that one should actually always use the same redis instance, right? As the fiat price conversions are always the same independent of the chain.